tests/codegen/simd-intrinsic/simd-intrinsic-mask-reduce.rs

```diff
@@ -0,0 +1,65 @@
+// compile-flags: -C no-prepopulate-passes
+//
+
+#![crate_type = "lib"]
+#![feature(repr_simd, platform_intrinsics)]
+#![allow(non_camel_case_types)]
+
+#[repr(simd)]
+#[derive(Copy, Clone)]
+pub struct mask32x2(i32, i32);
+
+#[repr(simd)]
+#[derive(Copy, Clone)]
+pub struct mask8x16(i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8, i8);
+
+extern "platform-intrinsic" {
+    fn simd_reduce_all<T>(x: T) -> bool;
+    fn simd_reduce_any<T>(x: T) -> bool;
+}
+
+// NOTE(eddyb) `%{{x|1}}` is used because on some targets (e.g. WASM)
+// SIMD vectors are passed directly, resulting in `%x` being a vector,
+// while on others they're passed indirectly, resulting in `%x` being
+// a pointer to a vector, and `%1` a vector loaded from that pointer.
+// This is controlled by the target spec option `simd_types_indirect`.
```

Comment:
This seems somewhat fickle, but I guess there isn't much in terms of alternatives here…

Reply:
This comment and logic was copied from another codegen test [simd-intrinsic-generic-bitmask.rs](https://github.com/rust-lang/rust/blob/master/tests/codegen/simd-intrinsic/simd-intrinsic-generic-bitmask.rs), Since the code there has not changed I assume there is still no better alternative.